core/indexer/process/accounts/accountsProcessor.go

```diff
@@ -0,0 +1,193 @@
+		balanceAsFloat := ap.computeBalanceAsFloat(userAccount.GetBalance())
+		acc := &types.AccountInfo{
+			Nonce:      userAccount.GetNonce(),
+			Balance:    userAccount.GetBalance().String(),
+			BalanceNum: balanceAsFloat,
+		}
+		address := ap.addressPubkeyConverter.Encode(userAccount.AddressBytes())
+		accountsMap[address] = acc
+	}
+
+	return accountsMap
+}
+
+// PrepareAccountsMapESDT will preaparare a map of accounts with ESDT tokens
+func (ap *AccountsProcessor) PrepareAccountsMapESDT(accounts []*WrappedUserAccount) map[string]*types.AccountInfo {
+	accountsESDTMap := make(map[string]*types.AccountInfo)
+	for _, wrappedUserAccount := range accounts {
+		address := ap.addressPubkeyConverter.Encode(wrappedUserAccount.Account.AddressBytes())
+
+		balance, properties, err := ap.getESDTInfo(wrappedUserAccount)
+		if err != nil {
+			log.Warn("cannot get esdt info from account",
+				"address", address,
+				"error", err.Error())
+			continue
+		}
+
+		acc := &types.AccountInfo{
+			Address:         address,
+			TokenIdentifier: wrappedUserAccount.TokenIdentifier,
+			Balance:         balance.String(),
+			BalanceNum:      ap.computeBalanceAsFloat(balance),
+			Properties:      properties,
+		}
+
+		accountsESDTMap[address] = acc
+	}
+
+	return accountsESDTMap
+}
+
+// PrepareAccountsHistory will prepare a map of accounts history balance form a map of accounts
+func (ap *AccountsProcessor) PrepareAccountsHistory(accounts map[string]*types.AccountInfo) map[string]*types.AccountBalanceHistory {
+	currentTimestamp := time.Now().Unix()
+	accountsMap := make(map[string]*types.AccountBalanceHistory)
+	for address, userAccount := range accounts {
+		acc := &types.AccountBalanceHistory{
+			Address:         address,
+			Balance:         userAccount.Balance,
+			Timestamp:       currentTimestamp,
+			TokenIdentifier: userAccount.TokenIdentifier,
+		}
+		addressKey := fmt.Sprintf("%s_%d", address, currentTimestamp)
+		accountsMap[addressKey] = acc
+	}
+
+	return accountsMap
+}
+
+func (ap *AccountsProcessor) getESDTInfo(wrappedUserAccount *WrappedUserAccount) (*big.Int, string, error) {
+	if wrappedUserAccount.TokenIdentifier == "" {
+		return nil, "", nil
+	}
+
+	tokenKey := core.ElrondProtectedKeyPrefix + core.ESDTKeyIdentifier + wrappedUserAccount.TokenIdentifier
+	valueBytes, err := wrappedUserAccount.Account.DataTrieTracker().RetrieveValue([]byte(tokenKey))
+	if err != nil {
+		return nil, "", nil
+	}
+
+	esdtToken := &esdt.ESDigitalToken{}
+	err = ap.internalMarshalizer.Unmarshal(esdtToken, valueBytes)
+	if err != nil {
+		return nil, "", nil
+	}
+
+	return esdtToken.Value, hex.EncodeToString(esdtToken.Properties), nil
+}
+
+func (ap *AccountsProcessor) computeBalanceAsFloat(balance *big.Int) float64 {
+	balanceBigFloat := big.NewFloat(0).SetInt(balance)
+	balanceFloat64, _ := balanceBigFloat.Float64()
+
+	bal := balanceFloat64 / ap.dividerForDenomination
+	balanceFloatWithDecimals := math.Round(bal*ap.balancePrecision) / ap.balancePrecision
```

Comment:
is this correct ?

Reply:
@bogdan-rosianu can you confirm why you use this ? I remember that you have implemented this part of code.

Reply:
yes. it's the way you can display double numbers with the number of decimals needed. like `10.00` with 2 decimals